lib/core/src/Cardano/Pool/DB/Sqlite.hs

```diff
@@ -208,14 +211,24 @@ newDBLayer trace fp timeInterpreter = do
 
             pure (foldl' toMap Map.empty production)
 
-        readTotalProduction = do
-            production <- fmap entityVal <$>
-                selectList ([] :: [Filter PoolProduction]) []
-
-            let toMap m (PoolProduction{poolProductionPoolId}) =
-                    Map.insertWith (+) poolProductionPoolId 1 m
-
-            pure $ Map.map Quantity $ foldl' toMap Map.empty production
+        readTotalProduction =
+            Map.fromList . fmap (assertSuccess . parseRow) <$> rawSql query []
+          where
+            assertSuccess = flip either id $ \e -> error $ mconcat
+                [ "readTotalProduction: unexpected parse failure: "
+                , show e
+                ]
```

Comment:
Since we're in IO it's best to throw it as IO errors. `error` triggers on evaluation and can leak through exception handlers. Probably not a big concern here, but better safe than sorry.

```suggestion
        readTotalProduction = do
            rows <- rawSql query []
            parsed <- forM rows $ \row -> case parseRow row of
                Right x -> pure x
                Left e  -> liftIO . throwIO . userError
                    . ("readTotalProduction: unexpected parse failure: " ++)
                    . T.unpack $ e
            pure $ Map.fromList parsed
          where
```

Reply:
Thanks @hasufell , I agree it's better to record parse failures eagerly, as soon as they happen, rather than relying on the lazy evaluation of `error`.

In the end I changed this to use the logging mechanism instead, as I think this will make any errors easier to analyze.

Reply:
Logging doesn't short-circuit and doesn't crash the handler, though. From what I remember, we don't expect parse failures here ever?